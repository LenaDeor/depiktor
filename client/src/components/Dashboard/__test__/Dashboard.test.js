import React from 'react';
import ReactDOM from 'react-dom';
import Dashboard from './../Dashboard';

import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";


it ("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Dashboard></Dashboard>, div)
})

it ("matches snapshot", () => {
  const tree = renderer.create(<Dashboard label="save"></Dashboard>).toJSON();
  expect(tree).toMatchSnapshot();
})
