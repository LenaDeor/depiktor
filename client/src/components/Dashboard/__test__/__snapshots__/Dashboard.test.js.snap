// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="dashboard"
>
  <div
    className="options-container"
  >
    <div
      className="chart-options-container"
    >
      <div
        className="options-header"
      >
        <p>
          Options
        </p>
      </div>
      <div
        className="chart-style-container"
      >
        <p>
          Chart Style
        </p>
        <div
          className="chart-style-select-dropdown css-2b097c-container"
          onKeyDown={[Function]}
        >
          <div
            className=" css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-g1d714-ValueContainer"
            >
              <div
                className=" css-1uccc91-singleValue"
              >
                Line
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-3-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-6q0nyr-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="chart-timeframe-container"
      >
        <p>
          Timeframe
        </p>
        <div
          className="horizontal-slider"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="options-track options-track-0"
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "right": 0,
                "willChange": "",
              }
            }
          />
          <div
            className="options-track options-track-1"
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "right": 0,
                "willChange": "",
              }
            }
          />
          <div
            aria-orientation="horizontal"
            aria-valuemax={0}
            aria-valuemin={0}
            aria-valuenow={0}
            className="options-thumb options-thumb-0 "
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            role="slider"
            style={
              Object {
                "left": "0px",
                "position": "absolute",
                "touchAction": "none",
                "willChange": "",
                "zIndex": 1,
              }
            }
            tabIndex={0}
          >
            0
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="tabs-container"
  >
    <div
      className="spinner"
    >
      <div
        className="rect1"
      />
      <div
        className="rect2"
      />
      <div
        className="rect3"
      />
      <div
        className="rect4"
      />
      <div
        className="rect5"
      />
    </div>
  </div>
</div>
`;
